test/pg-cdc/pg-cdc.td

```diff
@@ -0,0 +1,392 @@
+ALTER TABLE nonpk_table REPLICA IDENTITY FULL;
+INSERT INTO nonpk_table VALUES (2, 2), (2, 2);
+
+CREATE TABLE types_table (date_col DATE, time_col TIME, timestamp_col TIMESTAMP, uuid_col UUID, double_col DOUBLE PRECISION);
+INSERT INTO types_table VALUES ('2011-11-11', '11:11:11', '2011-11-11 11:11:11', 'A0EEBC99-9C0B-4EF8-BB6D-6BB9BD380A11', 1234.56768);
+ALTER TABLE types_table REPLICA IDENTITY FULL;
+
+CREATE TABLE large_text (f1 TEXT);
+INSERT INTO large_text VALUES (REPEAT('x', 16 * 1024 * 1024));
+ALTER TABLE large_text REPLICA IDENTITY FULL;
+
+CREATE TABLE trailing_space_pk (f1 TEXT PRIMARY KEY);
+INSERT INTO trailing_space_pk VALUES ('abc   ');
+ALTER TABLE trailing_space_pk REPLICA IDENTITY FULL;
+
+CREATE TABLE trailing_space_nopk (f1 TEXT);
+INSERT INTO trailing_space_nopk VALUES ('abc   ');
+ALTER TABLE trailing_space_nopk REPLICA IDENTITY FULL;
+
+CREATE TABLE multipart_pk(f1 INTEGER, f2 TEXT, f3 TEXT, PRIMARY KEY (f1, f2));
+INSERT INTO multipart_pk VALUES (1, 'abc', 'xyz');
+ALTER TABLE multipart_pk REPLICA IDENTITY FULL;
+
+CREATE TABLE nulls_table (f1 TEXT, f2 INTEGER);
+INSERT INTO nulls_table VALUES (NULL, NULL);
+ALTER TABLE nulls_table REPLICA IDENTITY FULL;
+
+CREATE TABLE utf8_table (f1 TEXT PRIMARY KEY, f2 TEXT);
+INSERT INTO utf8_table VALUES ('това е текст', 'това ''е'' "текст"');
+ALTER TABLE utf8_table REPLICA IDENTITY FULL;
+
+CREATE TABLE no_replica_identity (f1 INTEGER);
+INSERT INTO no_replica_identity VALUES (1), (2);
+
+#
+# Error checking
+#
+
+> CREATE MATERIALIZED SOURCE "no_such_host"
+  FROM POSTGRES HOST 'host=no_such_postgres.mtrlz.com port=5432 user=postgres password=postgres dbname=postgres'
+  PUBLICATION 'mz_source' NAMESPACE 'public' TABLE 'pk_table' (f1 TIMESTAMP, f2 TEXT);
+
+! SELECT * FROM no_such_host;
+Source error: File IO: error connecting to server: failed to lookup address information: Name or service not known
+
+> CREATE MATERIALIZED SOURCE "no_such_port"
+  FROM POSTGRES HOST 'host=postgres port=65534 user=postgres password=postgres dbname=postgres'
+  PUBLICATION 'mz_source' NAMESPACE 'public' TABLE 'pk_table' (f1 TIMESTAMP, f2 TEXT);
+
+! SELECT * FROM no_such_port;
+Source error: File IO: error connecting to server: Connection refused (os error 111)
+
+> CREATE MATERIALIZED SOURCE "no_such_user"
+  FROM POSTGRES HOST 'host=postgres port=5432 user=no_such_user password=postgres dbname=postgres'
+  PUBLICATION 'mz_source' NAMESPACE 'public' TABLE 'pk_table' (f1 TIMESTAMP, f2 TEXT);
+
+! SELECT * FROM no_such_user;
+Source error: File IO: db error: FATAL: role "no_such_user" does not exist
+
+> CREATE MATERIALIZED SOURCE "no_such_password"
+  FROM POSTGRES HOST 'host=postgres port=5432 user=postgres password=no_such_password dbname=postgres'
+  PUBLICATION 'mz_source' NAMESPACE 'public' TABLE 'pk_table' (f1 TIMESTAMP, f2 TEXT);
+
+! SELECT * FROM no_such_password;
+internal error: column 0 is not of expected type ColumnType { nullable: true, scalar_type: Timestamp }: Int32(1)
```

Comment:
This is the error for "wrong password". I think it comes from the fact that the pg client gets unceremoneously disconnected.